platform/core/src/services/HangingProtocolService/HangingProtocolService.js

```diff
@@ -387,7 +421,7 @@ class HangingProtocolService {
     );
 
     const bestMatch = matchingScores[0];
-
+    if (bestMatch?.displaySet) existingMatches.push(bestMatch.displaySet);
```

Comment:
If for any reason we end up using the existingMatches (which I explained why we shouldn't above), this push to array should happen outside of `match` function